pkg/pgmodel/querier/series_set.go

```diff
@@ -86,6 +87,17 @@ func (p *pgxSeriesSet) At() storage.Series {
 			log.Error("err", err)
 			return nil
 		}
+
+		if schemaLabelName, schemaLabelValue := getSchemaLabel(row.schema); schemaLabelName != "" {
```

Comment:
A similar code is repeated in `buildTimeSeries()`.  Can/Should we abstract this down to the function that is responsible for this task? IMHO, It will look the code clean, otherwise its more branches at the moment.

Reply:
It would be nice but because we are dealing with different structs (labels.Label vs. prompb.Label) its not really easy to reuse the code. Its possible but it would add complexity to the code.